### ISSUE TITLE
Add a timeout to AnalyzeAPK

### ISSUE THREAD

**Issue by projectbtle** [androguard/androguard#736]
Would it be possible to add a timeout value to the `AnalyzeAPK` function, so that when it is called from a single-threaded application, it can be killed if it takes too long? 

I have a script that is supposed to process thousands of APKs, and for some reason, `AnalyzeAPK` gets stuck on one particular APK (it had been processing the APK for 7 hours before I killed the script).  I could exclude the known bad APK, but there may be others. 

Right now, the solution is probably to re-do the script to AnalyeAPK in a separate thread that can be terminated on timeout. But it would be nice if the timeout could be an argument to `AnalyzeAPK` itself.

**Comment by reox** [comment]
This sounds like a good idea! The problem is, obviously, you might get incorrect xref results.
Btw can you share the apk? It would be interessting to know if it triggers any endless loops...

**Comment by projectbtle** [comment]
For my use case, I would just discard the result altogether, but yes, if someone was to try and make use of the result, it might be problematic.

I'll get the APK from our server and share it (I just have to double-check that it's ok first).

**Comment by reox** [comment]
Okay, yes discarding the results would probably be the best.
If you want to implement this cheaply on your side, you can probably use https://docs.python.org/3/library/threading.html#threading.Thread and some semaphore with a timeout.

**Comment by projectbtle** [comment]
Yes, I used `multiprocessing.Process` and called `terminate()` on timeout.

Unfortunately, the server load has now greatly reduced and I am not able to reproduce the issue any more (this is despite having previously repeated the test multiple times to have it always hang on the same APK). Nevertheless, I have attached the file that I _believe_ was the problematic one.

Thanks!

[timeout.zip](https://github.com/androguard/androguard/files/3649551/timeout.zip)
